### DISCUSSION TITLE
Install Oqtane using Sqlite database

### DISCUSSION THREAD

**Discussion by medboz** [oqtane/oqtane.framework#851] [General]
Hello Oqtane community,

I'm a begginer with the Oqtane framework. When I tried it, I saw that the installation page gives two options for the database provider(localdb or SqlServer). Is it planned to support Sqlite database in Oqtane ?

**Comment by sbwalker** [comment]
It is on the roadmap, but will not be in the 2.0 release. I assume you are asking because you want to use Oqtane on Linux?

**Reply by medboz** [reply]
Hello @sbwalker 
Thank you for the answer. Yes, I'd like to use Oqtane in a personal project, so I don't want to use SqlServer and as you said I'm planning to use linux.
Do you recommend me to fork Octane solution and use Sqlite in the Oqtane DbContext?

**Reply by hishamco** [reply]
Nothing stop you @medboz, while EF Core supports SQLite

**Reply by medboz** [reply]
thank you @hishamco

**Reply by sbwalker** [reply]
@medboz the main challenge you will run into is the installer relies on DBUp rather than DB migrations. DBUp is a script-based solution for provisioning/upgrading databases and currently Oqtane only supports SQL Server scripts. So you will need to figure out how to manually provision a SQLite database. Once the database is created, you should be able to switch the references from the EFCore SQL Server provider to the EFCore SQLite provider and it should work. However you will not be able to use some external modules as they also rely on SQL server. As you can see this is a larger problem which is why it has not yet been tackled in the core framework and remains on the roadmap for the future.

**Reply by medboz** [reply]
@sbwalker: It seems more complicated then. In that case I will wait the sqlite provider. In the meantime, I will probably find a way to run it using SQL server.
